citations/citations-impl/impl/src/java/org/sakaiproject/citation/impl/BaseCitationService.java

```diff
@@ -826,7 +828,22 @@ public void exportRis(StringBuilder buffer) throws IOException
 
 			// TODO: deal with real dates. Right now, just year
 
-  		exportRisField("Y1", getCitationProperty(Schema.YEAR, false) + "//", buffer);
+			exportRisField("Y1", getCitationProperty(Schema.YEAR, false) + "//", buffer);
+
+			// Extract the LCN from the otherIds citation field (if populated) and put in RIS U3.
+			String property = (String) getCitationProperty("otherIds", true);
+			if (property != null || property.length() > 0) {
+				int position = 0;
+				if ((position = property.indexOf("doc=oxfaleph")) > 0) {
+					String oxfalephId = property.substring(position + "doc=oxfaleph".length(), property.indexOf('&', position + 1));
```

Comment:
This will error out with `IllegalArgumentException` I think if there isn't a `&` following the `doc=oxfaleph`

Reply:
The javadoc seems to suggest a -1 would be returned in this case, however I've changed to code so the substring still works.